### PR TITLE
Fixing triggerGrinder.groovy. 

### LINKED ISSUE #2702: Read json file to trigger Jenkins Grinder
As part of [Create a Disabled tests pipeline](https://github.com/adoptium/aqa-tests/issues/1853), we should read this JSON file and trigger Grinder for GIT_ISSUE_STATUS=closed

- input file:
```
[{
"JDK_VERSION" : 17,
"JDK_IMPL" : "hotspot",
"TARGET" : "jdk_custom",
"CUSTOM_TARGET" : "java/lang/ClassLoader/nativeLibrary/NativeLibraryTest.java",
"GIT_ISSUE" : "https://github.com/adoptium/temurin-build/issues/248",
"PLATFORM" : "all",
"GIT_ISSUE_STATUS": "closed"
},{
"JDK_VERSION" : 16,
"JDK_IMPL" : "openj9",
"TARGET" : "jdk_custom",
"CUSTOM_TARGET" : "java/lang/ClassLoader/Assert.java",
"GIT_ISSUE" : "https://github.com/eclipse-openj9/openj9/issues/6668",
"PLATFORM" : "x86-64_mac",
"GIT_ISSUE_STATUS": "open"
}]
```

- trigger Grinder for GIT_ISSUE_STATUS=closed. This can be a separate Jenkins job that does the following:
  - read JSON file in groovy
  - find the records with GIT_ISSUE_STATUS=closed
  - trigger [Grinder](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/) with the parameters
  - optional: this Jenkins job can take JSON as a parameter instead of reading a file on disk

### PR DESCRIPTION
**Pull request by amir-waheed**
The tests are now started without the use of URL. 
Fixes #2702. 

Signed-off-by: Abeer Waheed <amir2@ualberta.ca>

### CODE REVIEW

File: buildenv/jenkins/triggerGrinder.groovy [hunk 1]
```diff
@@ -1,38 +1,27 @@
 #!groovy
 
-def json_path = args.size() ? args[0] : "disabledTestParser/output.json."
+def json_path = args.size() ? args[0] : "disabledTestParser/output.json"
```
Comment (llxia) [inline buildenv/jenkins/triggerGrinder.groovy:3]: Plz change it to `def json_path = "disabledTestParser/output.json"`

Comment (amir-waheed) [reply buildenv/jenkins/triggerGrinder.groovy]: Done.

File: buildenv/jenkins/triggerGrinder.groovy [hunk 2]
```diff
@@ -1,46 +1,40 @@
 #!groovy
-
-def json_path = args.size() ? args[0] : "disabledTestParser/output.json."
-def json = readJSON file: json_path
-def jenkins_url = "https://ci.adoptopenjdk.net/job/Grinder/buildWithParameters"
-
+node {
+  stage('readJSON')
```
Comment (sophia-guo) [inline buildenv/jenkins/triggerGrinder.groovy:3]: Before the jenkins job, `checkout scm` step is needed, which checkout code from source control. https://www.jenkins.io/doc/pipeline/steps/workflow-scm-step/

Comment (amir-waheed) [reply buildenv/jenkins/triggerGrinder.groovy:12]: @sophia-guo thank you for the suggestion. The test now passes.

**Comment by amir-waheed** [comment]
@smlambert The test Is finally passing. Here is the link to the test: https://ci.adoptopenjdk.net/view/Test_grinder/job/AutoReenableJob/53/console
Is there anything else I need to do for this issue?

File: buildenv/jenkins/triggerGrinder.groovy [hunk 3]
```diff
@@ -1,46 +1,49 @@
...
-}
-
-/**
- Get each of the key-value pairs for each json value.
- If the issue for a job is closed, run grinder on it.
- Otherwise ignore it */
-json.eachWithIndex { dict, _ ->
-  if (dict["GIT_ISSUE_STATUS"] == "closed") {
-    run_grinder(dict)
+    build job: Grinder, parameters: childParams, propagate: false
```
Comment (smlambert) [inline buildenv/jenkins/triggerGrinder.groovy:47]: This line sets up and triggers a Grinder job to be run.

We need extra information to verify that this Grinder actually got triggered.  @amir-waheed - Can we print out the information relating to it, so we can more easily find it to see how it behaved.

**Comment by amir-waheed** [comment]
@smlambert @llxia I have now cleaned the code so I can be merged. I have also stored the result of the job in a variable called isSuccess. Following the link to the test I ran:
https://ci.adoptopenjdk.net/view/Test_grinder/job/AutoReenableJob/60/

File: buildenv/jenkins/triggerGrinder.groovy [hunk 4]
```diff
@@ -1,46 +1,49 @@
...
- If the issue for a job is closed, run grinder on it.
- Otherwise ignore it */
-json.eachWithIndex { dict, _ ->
-  if (dict["GIT_ISSUE_STATUS"] == "closed") {
-    run_grinder(dict)
+  for (e in childParams) {
+    println e
   }
+
+  def job = build job: "Grinder", parameters: childParams, propagate: false
```
Comment (llxia) [inline buildenv/jenkins/triggerGrinder.groovy:46]: Could you change it to `propagate: true`? And I think we do not need lines 47 and 48.

**Review by llxia — APPROVED** [review]
LGTM

**Comment by llxia** [comment]
Thanks @amir-waheed . We will have follow-up PR to improve this.

**Review by smlambert — APPROVED** [review]
thanks @amir-waheed !

**Pull request merged**